avro/avro_test.go

```diff
@@ -4,179 +4,342 @@ import (
 	})
 }
 
-func createMessage(schema string) ([]byte, error) {
-	marshalSchema := &Schema{
-		Definition: testSchema1,
-	}
+// Test marshalStringSlice function
+func TestUnitMarshalStringSlice(t *testing.T) {
+	Convey("Successfully marshal string slice", t, func() {
+		_, v, _ := setUp(testArraySchema, 3)
 
-	data := &testData1{
-		Manager:         "John Elway",
-		TeamName:        "Denver Broncos",
-		Owner:           "Pat Bowlen",
-		Sport:           "American Football",
-		HasChangedName:  false,
-		Uri:             "http://denverbroncos.com",
-		NumberOfPlayers: 11,
-	}
+		slice := marshalStringSlice(v, 0)
+		So(slice, ShouldNotBeEmpty)
+		So(slice, ShouldResemble, []interface{}{"1934", "1972", "1999"})
+	})
+}
+
+// Test marshalStructSlice function
+func TestUnitMarshalStructSlice(t *testing.T) {
+	Convey("Successfully marshal struct slice", t, func() {
+		avroSchema, v, _ := setUp(testNestedArraySchema, 4)
+
+		slice, err := marshalStructSlice(v, 1, avroSchema, "footballers")
+		So(err, ShouldBeNil)
+		So(slice, ShouldNotBeEmpty)
+	})
+
+	Convey("Throws error due to missing nested schema", t, func() {
+		avroSchema, v, _ := setUp(testNestedArraySchema, 4)
+
+		slice, err := marshalStructSlice(v, 1, avroSchema, "footballer")
+		So(err, ShouldNotBeNil)
+		So(err, ShouldResemble, ErrMissingNestedScema)
+		So(slice, ShouldBeNil)
+	})
+}
+
+// Test getArraySchema function
+func TestUnitGetArraySchema(t *testing.T) {
+	avroSchema, _, _ := setUp(testSchema, 0)
+
+	Convey("Successfully retrieve array schema", t, func() {
+		newSchema, err := getArraySchema(avroSchema, "manager")
+		So(err, ShouldBeNil)
+		So(newSchema, ShouldNotBeEmpty)
+	})
+
+	Convey("Error retrieving array schema", t, func() {
+		newSchema, err := getArraySchema(avroSchema, "managers")
+		So(err, ShouldNotBeNil)
+		So(err, ShouldResemble, ErrMissingNestedScema)
+		So(newSchema, ShouldBeNil)
+	})
+}
+
+// Test getNestedSchema function
+func TestUnitGetNestedSchema(t *testing.T) {
+	Convey("Successfully retrieve nested schema", t, func() {
+		avroSchema, v, t := setUp(testNestedArraySchema, 4)
+
+		newSchema, err := getNestedSchema(avroSchema, "footballers", v, t)
+		So(err, ShouldBeNil)
+		So(newSchema, ShouldNotBeEmpty)
+	})
+
+	Convey("Error retrieving nested schema", t, func() {
+		avroSchema, v, t := setUp(testNestedArraySchema, 4)
+
+		newSchema, err := getNestedSchema(avroSchema, "footballer", v, t)
+		So(err, ShouldNotBeNil)
+		So(err, ShouldResemble, ErrMissingNestedScema)
+		So(newSchema, ShouldBeNil)
+	})
+}
+
+// TODO Test Unmarshal function
+
+// TODO -- Test populateStructFromSchema function
+
+// TODO -- -- Test generateDecodedRecord function
 
-	message, err := marshalSchema.Marshal(data)
-	if err != nil {
-		return nil, err
+// TODO -- -- Test unmarshalStringSlice function
+
+// TODO -- -- Test unmarshalStructSlice function
+
+// TODO -- -- -- Test populateNestedArrayItem function
+
+// TODO -- -- -- -- Test setNestedStructs function
```

Comment:
There a section of TODO comments here. looks like you done the tests for them. Can they be removed?